APM/CommandLoggerRegistry.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\Bundle\MongoDBBundle\APM;
+
+use Doctrine\ODM\MongoDB\APM\CommandLoggerInterface;
+use function array_map;
+
+final class CommandLoggerRegistry
+{
+    /** @var CommandLoggerInterface[] */
+    private $commandLoggers = [];
+
+    public function __construct(iterable $commandLoggers)
+    {
+        foreach ($commandLoggers as $commandLogger) {
+            $this->addLogger($commandLogger);
+        }
+    }
+
+    public function register() : void
+    {
+        array_map(static function (CommandLoggerInterface $commandLogger) {
+            $commandLogger->register();
+        }, $this->commandLoggers);
+    }
+
+    public function unregister() : void
+    {
+        array_map(static function (CommandLoggerInterface $commandLogger) {
```

Comment:
Foreach is bad? :D

Reply:
One benefit is that this applies type checking to each element, which would otherwise require an `instanceof` check within `foreach`.

Reply:
That was one of the reasons why I added it. Of course, a type check is already done in the constructor when storing the items, but then I've started using `array_map` in place of `foreach` to make it more explicit. I can change it to a loop if you prefer that